src/libraries/System.Private.CoreLib/src/System/Environment.SunOS.cs

```diff
@@ -2,12 +2,36 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
 
 namespace System
 {
     public static partial class Environment
     {
         public static long WorkingSet =>
             (long)(Interop.procfs.TryReadProcessStatusInfo(Interop.procfs.ProcPid.Self, out Interop.procfs.ProcessStatusInfo status) ? status.ResidentSetSize : 0);
+
+        /// <summary>
+        /// Get the CPU usage, including the process time spent running the application code, the process time spent running the operating system code,
+        /// and the total time spent running both the application and operating system code.
+        /// </summary>
+        [SupportedOSPlatform("maccatalyst")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        public static ProcessCpuUsage CpuUsage
+        {
+            get
+            {
+                Interop.procfs.ParsedStat stat = GetStat();
+                return new ProcessCpuUsage { UserTime = TicksToTimeSpan(GetStat().Utime), PrivilegedTime = TicksToTimeSpan(GetStat().Stime) };
```

Comment:
Thanks. I opened the PR as a draft as it is not ready yet for review. I want to see if it will compile and pass tests first in all OSs. You can continue commenting though if you like :-)

Reply:
We have used other ways implementing it.